Add request logging to fake server

aburgel opened the pull request:
/domain @Betterment/test_track_core 
/platform @jmileham 

I added some very basic logging to make testing a little easier. It's disabled by default and looks like:

`2019/06/14 11:44:40 127.0.0.1:53261 GET /api/v2/split_registry`

aburgel commented on fakeserver/server.go:
```diff
@@ -24,20 +22,15 @@ type server struct {
-func Start(listenOn string) {
-	var wait time.Duration
-	flag.DurationVar(&wait, "graceful-timeout", time.Second*15, "the duration for which the server gracefully wait for existing connections to finish - e.g. 15s or 1m")
-	flag.Parse()
```
I removed this flag stuff since it was unused.

jmileham replied:
Thanks for catching.

aburgel commented on fakeserver/server.go:
```diff
@@ -64,7 +57,23 @@ func Start(listenOn string) {
+	if logRequests {
+		r.Use(loggingMiddleware)
+	}
+	r.Use(corsMiddleware)
```
refactored to use mux's `Use` functionality which allows you to apply middleware handlers

samandmoore commented:
seems great. i dunno how i feel about the command line flag name, but i'm not coming up with anything better.

<<domainlgtm

aburgel commented:
> i dunno how i feel about the command line flag name, but i'm not coming up with anything better.

`--verbose`?

jmileham commented on cmds/server.go:
```diff
@@ -11,11 +11,13 @@ and nonsense.
 
 func init() {
 	serverCmd.Flags().StringVarP(&listenOn, "listen", "l", defaultListenOn, "IP address and port to listen on")
+	serverCmd.Flags().BoolVarP(&logRequests, "log-requests", "", false, "Log requests to stderr")
```
Should we just go ahead and always log? I just didn't want to solve this because I was lazy but I'm not sure it's inappropriate to log all the time. But second question - should we log to STDOUT, since that's a 12-factor standard?

aburgel replied:
i was thinking about doing just that, but then wasn't sure what to do with the logs when we run via launchd. do you know how to set that up?

aburgel replied:
and yes, we can change to stdout to follow 12-factor.

jmileham replied:
Not exactly, but I'm OK with dropping them on the floor for starters and iterating toward doing it better over time. The real value so far has proven to be interactive testing anyway.

jmileham commented on fakeserver/server.go:
```diff
@@ -64,7 +57,23 @@ func Start(listenOn string) {
+	}
+	r.Use(corsMiddleware)
+
+	fmt.Printf("testtrack server listening on %s\n", listenOn)
```
We might want to make this a standard log message if we turn logging on all the time.

jmileham commented on fakeserver/server.go:
```diff
@@ -24,20 +22,15 @@ type server struct {
-	s.routes()
+func loggingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Println(r.RemoteAddr, r.Method, r.RequestURI)
```
Does this auto-flush by default? 12-factor also suggests you shouldn't buffer stdout log output.

aburgel replied:
yep, logging to stdout is unbuffered by default

jmileham commented on fakeserver/server.go:
```diff
@@ -64,7 +57,23 @@ func Start(listenOn string) {
+	r.Use(corsMiddleware)
+
+	fmt.Printf("testtrack server listening on %s\n", listenOn)
+	log.Fatal(http.ListenAndServe(listenOn, r))
```
should we log this in standard format too?

jmileham commented:
<< platform tafn - looks like a rebase might be needed too.

aburgel commented:
bump. removed the flag for logging, its now always on.

i also moved the rest of the server module's logging to use a single logger that goes to stdout. some examples:

launch:
```
2019/06/18 13:22:11 testtrack server listening on 127.0.0.1:8298
```

success:
```
2019/06/18 13:22:26 127.0.0.1:55752 - GET /api/v1/split_registry
```

error:
```
2019/06/18 13:22:26 got unexpected content type
```

fatal:
```
2019/06/18 13:26:22 fatal - listen tcp 127.0.0.1:1: bind: permission denied
```

readable, tho not exactly parsable, but i think that's fine for our purposes. lmk if you think otherwise.

jmileham commented:
sounds great! platform lgtm
